### DISCUSSION TITLE
Grain reconstruction based on atomistic (Molecular dynamics) data

### DISCUSSION THREAD

**Discussion by aminsakic** [mtex-toolbox/mtex#1977] [Ask Anything]
Hello MTEX community,

I am relatively new to MTEX and would like to compare grain boundary (GB) misorientations and GB types between experimental EBSD measurements and a 3D polycrystal obtained from molecular dynamics (MD). So far, I have successfully characterized the experimental EBSD data. However, I am currently facing the challenge of reconstructing my MD data using MTEX, and I'm uncertain about how to proceed.

I have examined the polycrystal in OVITO software and utilized Polyhedral Template Matching (PTM) to extract local (per-atom) crystal orientations for each particle within the simulation box. My goal is to bring this MD data into MTEX for further analysis.

I am seeking guidance on whether it's possible to work with this PTM-MD data in MTEX and, if so, how I can approach this task. If you require any specific data or additional details, please feel free to ask, and I'd be more than willing to share them in this chat.

Best regards,

**Comment by kilir** [comment]
Hi,
so your data looks like points with x,y,z coordinates and orientations at those points and you'd like to cut a slice through that point cloud to get something like an orientation map?
Cheers,
Rüdiger

**Comment by aminsakic** [comment]
Hello and thank you for your interest!

I have collected x, y, and z data for each particle within the simulation cell, along with orientations provided by the PTM encoded as quaternions. I successfully transformed these orientations (quaternions) into Euler angles, as demonstrated in the screenshot of the file attached. Furthermore, I have included information about which grain each particle belongs to, indicated in the last column of the data table.

The 3D simulation cell of the polycrystal is also displayed here. As mentioned earlier, my primary focus lies in analyzing the grain boundary misorientation, GB types, and grain orientations.


![fcc_polycrystall_3D](https://github.com/mtex-toolbox/mtex/assets/102667524/94b2776e-e1ab-4184-9501-67be53496c99)
![fcc_poly_md](https://github.com/mtex-toolbox/mtex/assets/102667524/7b05ae77-e230-4356-96c7-e36900de9f89)
